src/catsRouter.js

```diff
@@ -1,164 +1,169 @@
-        req.object = objToCheck; 
-        next();
+        return newError; 
     }
-}
-
-// checks object values 
-const checkObjValues = (req, res, next) => { 
-    const objToCheck = req.body;      // { name: "", ageInYears: "", favouriteToy: ""}
-
-    for (let key in objToCheck) {
-        if (key === "ageInYears") {
-            if (isInvalidNum(objToCheck.ageInYears)) { 
-                console.log(`invalid age value: ${objToCheck.ageInYears}`);
-                return next(generateErr(objToCheck.ageInYears));
-            } 
+    
+    // checks if id is a valid shortid                          
+    isIdValid (req, res, next) {       
+        console.log('id being verified');
+
+        if (this.shortid.isValid(req.params.id)) {   
+            req.id = req.params.id;                                             
+            console.log('id verified');
+            next() 
         } else {
-            if (isInvalidString(objToCheck[key])) { 
-                console.log(`invalid ${key} value: ${objToCheck[key]}`);
-                return next(generateErr(objToCheck[key]));
-            } 
+            const newError =  new Error(`'${req.params.id}' is not a valid shortid`)
+            newError.status = 400;
+            next(newError); 
         };
     }
 
-    req.object = objToCheck;            
-    console.log('object values checked');     
-    next();
-}
-
-// GET route all 
-catsRouter.get('', (req, res, next) => {
-    const cats = CatRepository.getAllCats();  //should return summary object
-    res.json({"cats": cats}); 
-});
-
-// GET route by id 
-catsRouter.get('/:id', isIdNum, (req, res, next) => {
-    const foundCat = CatRepository.getCatById(req.id); 
-    if (foundCat) {
-        console.log('cat retrieved:' + foundCat);
-        res.send(foundCat);
-    } else {
-        const newError =  new Error(`cat id '${req.id}' not found in database`)
-        newError.status = 404;
-        return next(newError)
-    };
-});
-
-
-// POST route 
-catsRouter.post('', checkObjFormat, checkObjKeys, checkObjValues, (req, res, next) => {
-    const catWithId = CatRepository.addCat(req.object);
-    res.status(201).send(catWithId);  
-});
-
-
-// PUT route - allows user to add/update information by id
-catsRouter.put('/:id', isIdNum, checkObjFormat, checkObjKeys, checkObjValues, (req, res, next) => {
-    const isUpdated = CatRepository.updateCatById(req.id, req.object);             // [ {}, {}, {} ]
-    if (isUpdated) {
-        console.log(`cat id '${req.id}' successfully updated`)
-        res.send(isUpdated); // may need to use: res.send(JSON.stringify(updatedCat)); 
-    } else {
-        const newError = new Error(`cat id '${req.id}' not found in database`);
-        newError.status = 404; 
-        next(newError); 
+    // checks object data format
+    checkObjFormat (req, res, next) { 
+        const objectConstructor = ({}).constructor; // in strict mode, 'this' in functions is undefined. 
```

Comment:
it'd be great if you could link to a blog post or something on this - often this can be a helpful reminder when reading the code again.

Reply:
I keep all useful blog posts links in my own notes to avoid cluttering up the code: 
https://www.vojtechruzicka.com/javascript-this-keyword/

Reply:
It's good you're keeping notes - but in this particular case it'd probably be helpful to link off to documentation about what is happening in case there are any JS newbies (like me) reading the code.